src/dependency-container.ts

```diff
@@ -338,20 +339,28 @@ class InternalDependencyContainer implements DependencyContainer {
     const paramInfo = typeInfo.get(ctor);
 
     if (!paramInfo || paramInfo.length === 0) {
-      throw new Error(`TypeInfo not known for ${ctor}`);
+      throw new Error(`TypeInfo not known for "${ctor.name}"`);
     }
 
-    const params = paramInfo.map(param => {
-      if (isTokenDescriptor(param)) {
-        return param.multiple
-          ? this.resolveAll(param.token)
-          : this.resolve(param.token, context);
-      }
-      return this.resolve(param, context);
-    });
+    const params = paramInfo.map(this.resolveParams(context, ctor));
 
     return new ctor(...params);
   }
+
+  private resolveParams<T>(context: ResolutionContext, ctor: constructor<T>) {
+    return (param: any, idx: number) => {
```

Comment:
Can we use the specific types rather than `any`?

Reply:
Yes, but I need to remove `any` from `typeInfo` map.

Something like this https://github.com/microsoft/tsyringe/pull/88/commits/144e07fd6fe5b338aae37a72e4416674a490d6b1  it's ok?